CopilotKit/packages/runtime/src/lib/runtime/copilot-runtime.ts

```diff
@@ -461,13 +461,27 @@ export class CopilotRuntime {
     agents: Record<string, AbstractAgent>,
     tools: BasicAgentConfiguration["tools"],
   ): Record<string, AbstractAgent> {
-    const enrichedAgents = { ...agents };
-    // Add tools to all existing BasicAgents by mutating their internal config
-    for (const existingAgent of Object.values(enrichedAgents)) {
-      // TODO: fix this - `existingAgent.config` does not exist on type 'AbstractAgent'
-      const config = existingAgent.config ?? {};
-      const existingTools = config.tools ?? [];
-      config.tools = [...existingTools, ...tools];
+    if (!tools || tools.length === 0) {
+      return agents;
+    }
+
+    const enrichedAgents: Record<string, AbstractAgent> = { ...agents };
+
+    for (const [agentId, agent] of Object.entries(enrichedAgents)) {
+      if (!(agent instanceof BasicAgent)) {
+        continue;
+      }
```

Comment:
Agent could be either basic agent, or any other (langgraph, mastra etc) therefore this check is not necessary